libraries/cyclestreets-view/src/main/java/net/cyclestreets/util/Permissions.kt

```diff
@@ -106,25 +121,30 @@ private fun activityFromContext(initialContext: Context): Activity? {
 }
 
 // Request permissions: activity or fragment
-fun requestPermission(activity: Activity?, fragment: Fragment?, permission: String, request_code: Int) {
+private fun requestPermission(activity: Activity?, fragment: Fragment?, permission: String, requestCode: Int) {
     try {
         if (fragment != null) {
             Log.d(TAG, "Entering Fragment requestPermissions operation for permission $permission")
-            fragment.requestPermissions(arrayOf(permission), request_code)
+            fragment.requestPermissions(arrayOf(permission), requestCode)
         }
         else {
             Log.d(TAG, "Entering Activity requestPermissions operation for permission $permission")
-            if (activity != null) {
-                activity.requestPermissions(arrayOf(permission), request_code)
-            }
+            activity?.requestPermissions(arrayOf(permission), requestCode)
         }
     } catch (e: IllegalStateException) {
         Log.w(TAG, "Unable to request permission $permission from ${if (fragment != null) "fragment $fragment" else "activity $activity"}", e)
     }
 }
 
-fun requestPermissionsResultAction(grantResults: IntArray, grantResult: Int?, permission: String, action: () -> Unit) {
-    if (grantResults.isNotEmpty() && grantResult == PackageManager.PERMISSION_GRANTED) {
+/**
+ * Generic handler for permission Handle
+ *
+ * @param grantResult - either [PERMISSION_GRANTED] or [PERMISSION_DENIED]
+ * @param permission - the string identifier of the permission, e.g. [Manifest.permission.ACCESS_FINE_LOCATION]
+ * @param action - the closure indicating the action to be taken if the permission was granted
+ */
+fun requestPermissionsResultAction(grantResult: Int?, permission: String, action: () -> Unit) {
+    if (grantResult == PERMISSION_GRANTED) {
```

Comment:
Just to clarify, grantResults.isNotEmpty() is superfluous if grantResult == PackageManager.PERMISSION_GRANTED ?

Reply:
Every place `requestPermissionsResultAction` is called is within a loop iterating through `grantResults`.  So if `grantResults` were ever empty, this method would not be called.